src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Solution.cs

```diff
@@ -8,123 +8,96 @@ namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+        public bool RequiresConfirmation { get; set; }
 
         /// <summary>
-        /// Denotes which action will be performed, such as a specific API call or navigation link.
+        /// The URI of the target resource on which an action will be performed.
         /// </summary>
-        [JsonConverter(typeof(StringEnumConverter))]
-        public ActionType Action { get; set; }
+        public string ResourceUri { get; set; }
 
         /// <summary>
-        /// Free-form JSON-serializable arguments that can be sent as the body of the action request if a request
-        /// body is applicable.
+        /// Instructions that will be sent to the customer by support staff. Rendered as Markdown.
         /// </summary>
-        public Dictionary<string, object> ActionArgs { get; set; }
+        public string InternalInstructions { get; set; }
+
+        public SolutionTypeTag TypeTag { get; set; }
 
         /// <summary>
-        /// Adds a pre-written <see cref="Description"/> to the Solution. Takes priority over Description values.
+        /// This is set automatically.
+        /// Reads <see cref="OperationContext"/> to determine if an internal view should be rendered.
         /// </summary>
-        [JsonConverter(typeof(StringEnumConverter))]
-        public SolutionText? PremadeDescription { get; set; }
+        public bool IsInternal { get; set; } = false;
 
         /// <summary>
-        /// Adds pre-written <see cref="InternalInstructions"/> to the Solution. Takes priority over
-        /// InternalInstructions values.
+        /// This is set automatically.
+        /// Used to create a deep link back to the Solution's detector in case the customer is receiving
+        /// Solution information from support staff.
         /// </summary>
-        [JsonConverter(typeof(StringEnumConverter))]
-        public SolutionText? PremadeInstructions { get; set; }
+        public string DetectorId { get; set; }
 
         /// <summary>
-        /// Creates a Solution component. Prefer a pre-defined Solution component such as <see cref="Solution.Restart()"/>.
+        /// Set the ResourceUri and optionally set the ActionOptions for the Solution to use.
         /// </summary>
-        /// <param name="title">Solution title.</param>
-        /// <param name="resourceUri">The URI of the target resource on which an action will be performed.</param>
-        /// <param name="isInternal">Reads <see cref="OperationContext"/> to determine if an internal view should be rendered.</param>
-        /// <param name="action">Denotes which action will be performed, such as a specific API call or navigation link.</param>
-        public Solution(string title, string resourceUri, ActionType action)
+        // TODO: actionOptions has to merge with existing options in case detector author adds to solution author's options
+        public Solution Using(string resourceUri, Dictionary<string, object> actionOptions = null)
```

Comment:
have you added sample code documentation anywhere on how authors can write Solution?

Reply:
Not yet, I'm wondering where the best place would be for that since it will be entirely Gist-based.

Reply:
It's possible we could have an "example Gist" that shows Solution documentation, once @crazyxy adds sub-folders for Gists and we make a Solutions folder.
For now I'll just be incorporating simple Solutions into existing Detectors, so nobody should have to write their own Solutions for a short while